npm/private/npm_link_package.bzl

```diff
@@ -273,10 +275,11 @@ deps of npm_link_package_store must be in the same package.""" % (ctx.label.pack
                                 target_file = dep_ref_def_virtual_store_directory,
                             )
                             direct_files.append(dep_ref_dep_symlink)
+                            symlinks.update({dep_ref_dep_symlink_path: dep_ref_def_virtual_store_directory})
 
     direct_files = depset(direct = direct_files)
     files_depsets = [direct_files]
-    runfiles = ctx.runfiles(transitive_files = direct_files)
+    runfiles = ctx.runfiles(transitive_files = direct_files, symlinks = symlinks)
```

Comment:
this change is green in the repo, but when I merged this file to `main` briefly @gregmagolan saw a bunch of errors even after a `clean --expunge`

```
ERROR: /Users/greg/aspect/rules/js/rules_js/examples/npm_deps/BUILD.bazel:134:14: overwrote runfile node_modules/.aspect_rules_js/rollup@2.70.2/node_modules/fsevents, was symlink to bazel-out/darwin-fastbuild/bin/node_modules/.aspect_rules_js/fsevents@2.3.2/node_modules/fsevents, now symlink to bazel-out/darwin-fastbuild/bin/node_modules/.aspect_rules_js/rollup@2.70.2/node_modules/fsevents
ERROR: /Users/greg/aspect/rules/js/rules_js/examples/npm_deps/BUILD.bazel:128:7: overwrote runfile node_modules/.aspect_rules_js/rollup@2.70.2/node_modules/fsevents, was symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/fsevents@2.3.2/node_modules/fsevents, now symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/rollup@2.70.2/node_modules/fsevents
ERROR: /Users/greg/aspect/rules/js/rules_js/examples/webpack_cli/BUILD.bazel:28:16: overwrote runfile node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@discoveryjs/json-ext, was symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/@discoveryjs+json-ext@0.5.7/node_modules/@discoveryjs/json-ext, now symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@discoveryjs/json-ext
ERROR: /Users/greg/aspect/rules/js/rules_js/examples/webpack_cli/BUILD.bazel:28:16: overwrote runfile node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@webpack-cli/configtest, was symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/@webpack-cli+configtest@1.2.0_77l47gmqkrqiei5z7sbwz5iaj4/node_modules/@webpack-cli/configtest, now symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@webpack-cli/configtest
ERROR: /Users/greg/aspect/rules/js/rules_js/examples/webpack_cli/BUILD.bazel:28:16: overwrote runfile node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@webpack-cli/info, was symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/@webpack-cli+info@1.5.0_webpack-cli@4.10.0/node_modules/@webpack-cli/info, now symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@webpack-cli/info
ERROR: /Users/greg/aspect/rules/js/rules_js/examples/webpack_cli/BUILD.bazel:28:16: overwrote runfile node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@webpack-cli/serve, was symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/@webpack-cli+serve@1.7.0_webpack-cli@4.10.0/node_modules/@webpack-cli/serve, now symlink to bazel-out/darwin-opt-exec-2B5CBBC6/bin/node_modules/.aspect_rules_js/webpack-cli@4.10.0_webpack@5.73.0/node_modules/@webpack-cli/serve
```

so we have to figure that out before we can merge again...

Reply:
probably, the symlink provided as a file and the one provided via symlinks dict of run files collide. it's weird that bazel accepts both...